### PR TITLE
Improve unit test coverage for REST API menus endpoints. 

### LINKED ISSUE #34992: Improve unit test coverage for REST API menus endpoints.
#### Background
The menus endpoints, menu, menu locations and menu items were originally worked on by @spacedmonkey and developed in [another](https://github.com/WP-API/menus-endpoints) github repo. These endpoints, were merged into this plugin in https://github.com/WordPress/gutenberg/pull/20292. 

Now that the menus endpoints are being used for the menu screen. There have been a number of changes to the REST API endpoint over the last couple of months these changes include. 

#31656, #34673, #34888, #34835

Let's audit and improve PHPUnit test coverage for these endpoints.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Description
Improve unit test coverage for REST API menus endpoints. 

Bug fixes and tweaks to code to improve code. 

Final tweaks, before core merge. 

Fixes #34992

### CODE REVIEW

File: lib/class-wp-rest-menu-items-controller.php [hunk 1]
```diff
@@ -479,21 +479,23 @@ protected function prepare_item_for_database( $request ) {
 			}
 		}
 
+		$error = new WP_Error();
```
Comment (spacedmonkey) [inline lib/class-wp-rest-menu-items-controller.php:482]: This is not needed, but makes debugging easier, as you can see multiple errors in REST API response.

File: lib/class-wp-rest-menu-items-controller.php [hunk 2]
```diff
@@ -324,7 +324,7 @@ public function update_item( $request ) {
 	 * Deletes a single menu item.
 	 *
 	 * @param WP_REST_Request $request Full details about the request.
-	 * @return true|WP_Error True on success, or WP_Error object on failure.
+	 * @return WP_REST_Response|WP_Error True on success, or WP_Error object on failure.
```
Comment (spacedmonkey) [inline lib/class-wp-rest-menu-items-controller.php:327]: This return type was wrong.

File: lib/class-wp-rest-menu-items-controller.php [hunk 3]
```diff
@@ -504,27 +506,38 @@ protected function prepare_item_for_database( $request ) {
...
-				return new WP_Error( 'rest_url_required', __( 'URL required if menu item of type custom.', 'gutenberg' ), array( 'status' => 400 ) );
+				$error->add( 'rest_url_required', __( 'URL required if menu item of type custom.', 'gutenberg' ), array( 'status' => 400 ) );
 			}
 		}
 
 		// If menu item is type block, then content is required.
-		if ( 'block' === $prepared_nav_item['menu-item-type'] ) {
-			if ( empty( $prepared_nav_item['menu-item-content'] ) ) {
-				return new WP_Error( 'rest_content_required', __( 'Content required if menu item of type block.', 'gutenberg' ), array( 'status' => 400 ) );
+		if ( 'block' === $prepared_nav_item['menu-item-type'] && empty( $prepared_nav_item['menu-item-content'] ) ) {
```
Comment (spacedmonkey) [inline lib/class-wp-rest-menu-items-controller.php:524]: No need for two if statements here.

File: lib/class-wp-rest-menu-items-controller.php [hunk 4]
```diff
@@ -755,16 +760,20 @@ protected function prepare_links( $menu_item ) {
...
 			$path = rest_get_route_for_term( $menu_item->object_id );
+			$type = get_term_field( 'taxonomy', $menu_item->object_id );
 		}
 
-		if ( $path ) {
+		if ( $path && $type ) {
 			$links['https://api.w.org/object'][] = array(
 				'href'       => rest_url( $path ),
-				'post_type'  => $menu_item->type,
+				$key         => $type,
```
Comment (spacedmonkey) [inline lib/class-wp-rest-menu-items-controller.php:776]: This was incorrect, this is a bug fix.

File: lib/class-wp-rest-menu-items-controller.php [hunk 5]
```diff
@@ -755,16 +760,20 @@ protected function prepare_links( $menu_item ) {
...
 			$path = rest_get_route_for_post( $menu_item->object_id );
+			$type = get_post_type( $menu_item->object_id );
 		} elseif ( 'taxonomy' === $menu_item->type ) {
 			$path = rest_get_route_for_term( $menu_item->object_id );
+			$type = get_term_field( 'taxonomy', $menu_item->object_id );
 		}
 
-		if ( $path ) {
+		if ( $path && $type ) {
 			$links['https://api.w.org/object'][] = array(
```
Comment (spacedmonkey) [inline lib/class-wp-rest-menu-items-controller.php]: @TimothyBJacobs did you want to change this?

Comment (TimothyBJacobs) [reply lib/class-wp-rest-menu-items-controller.php]: Yeah, let's change the type to `menu-item-object`

**Comment by TimothyBJacobs** [comment]
Looks good, thanks. Yeah let's just adjust that menu item link relation.

File: lib/class-wp-rest-menu-items-controller.php (hunk 5 above)
Comment (spacedmonkey) [reply lib/class-wp-rest-menu-items-controller.php]: Fixed in https://github.com/WordPress/gutenberg/pull/36177/commits/dc23cdc7dfae8e8c102ef58a8c24444513fa7bf8

**Review by TimothyBJacobs — APPROVED** [review]

**Pull request merged**